test/python/transpiler/test_mappers.py

```diff
@@ -0,0 +1,280 @@
+  ``self.shots``, ``self.delta``. They are required for the regeneration of the
+  ground truth.
+* use the ``self.assertResult`` assertion for comparing for regeneration of the
+  ground truth.
+* explicitly set a unique ``name`` of the ``QuantumCircuit``, as it it used
+  for the name of the pickle file of the ground truth.
+
+For example::
+
+    def test_a_common_test(self):
+        self.count = {'000': 512, '110': 512}  # The expected count for this circuit
+        self.shots = 1024                      # Shots to run in the backend.
+        self.delta = 5                         # This is delta for the AlmostEqual during
+                                               # the count check
+        coupling_map = [[0, 1], [0, 2]]        # The coupling map for this specific test
+
+        qr = QuantumRegister(3, 'q')               #
+        cr = ClassicalRegister(3, 'c')             # Set the circuit to test
+        circuit = QuantumCircuit(qr, cr,           # and don't forget to put a name
+                                 name='some_name') # (it will be used to save the pickle
+        circuit.h(qr[1])                           #
+        circuit.cx(qr[1], qr[2])                   #
+        circuit.measure(qr, cr)                    #
+
+        result = transpile(circuit, self.create_backend(), coupling_map=coupling_map,
+                           pass_manager=self.create_passmanager(coupling_map))
+        self.assertResult(result, circuit.name)
+```
+"""
+
+# pylint: disable=redefined-builtin,attribute-defined-outside-init
+
+import unittest
+import pickle
+
+from qiskit import ClassicalRegister, QuantumRegister, QuantumCircuit, BasicAer, compile
+from qiskit.transpiler import PassManager, transpile
+from qiskit.transpiler.passes import BasicSwap, LookaheadSwap, StochasticSwap
+from qiskit.mapper import CouplingMap, Layout
+
+from qiskit.test import QiskitTestCase
+
+
+class CommonUtilitiesMixin:
+    """Utilities for meta testing.
+
+    Subclasses should redefine the ``pass_class`` argument, with a Swap Mapper
+    class.
+
+    Note: This class assumes that the subclass is also inheriting from
+    ``QiskitTestCase``, and it uses ``QiskitTestCase`` methods directly.
+    """
+
+    regenerate_expected = False
+    seed = 42
+    additional_args = {}
+    pass_class = None
+
+    def create_passmanager(self, coupling_map, initial_layout=None):
+        """Returns a PassManager using self.pass_class(coupling_map, initial_layout)"""
+        passmanager = PassManager(
+            self.pass_class(CouplingMap(coupling_map),  # pylint: disable=not-callable
+                            **self.additional_args))
+        if initial_layout:
+            passmanager.property_set['layout'] = Layout(initial_layout)
+        return passmanager
+
+    def create_backend(self):
+        """Returns a Backend."""
+        return BasicAer.get_backend('qasm_simulator')
+
+    def generate_expected(self, transpiled_result, filename):
```

Comment:
Non blocking: you have talked about the “ground truth” so I would incorporate that term to the API and this seems like a good place to do it. What about changing the name to: `generate_ground_truth`?

Reply:
done in 5dca49da